### PR TITLE
Add Bayesian NODE, Applications document

### PR DESCRIPTION
**Pull request by RajDandekar**

### CODE REVIEW

File: docs/src/getting_started/bayesian_neural_ode_introduction.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+function loss_neuralode(p)
+    pred = predict_neuralode(p)
+    loss = sum(abs2, ode_data .- pred)
+    return loss, pred
+end
+
+```
+
+## Step 4: Step 4: Now we start integrating the Bayesian estimation workflow as prescribed by the AdvancedHMC interface with the NeuralODE defined above
```
Comment (Vaibhavdixit02) [inline docs/src/getting_started/bayesian_neural_ode_introduction.md:54]:
```suggestion
## Step 4: Now we start integrating the Bayesian estimation workflow as prescribed by the AdvancedHMC interface with the NeuralODE defined above
```

File: docs/src/highlevels/applications.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+## Battery science
+
+- Universal battery performance and degradation model for electric aircraft (https://arxiv.org/abs/2008.01527)
+
+## Chemical science
+
+- Autonomous discovery of unknown reaction pathways from data by chemical reaction neural network (https://arxiv.org/pdf/2002.09062.pdf)
+
+## Pharmocometrics
```
Comment (Vaibhavdixit02) [inline docs/src/highlevels/applications.md:37]:
```suggestion
## Pharmacometrics
```

File: docs/src/getting_started/bayesian_neural_ode_introduction.md [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+
+The Advanced HMC interface requires us to specify: (a) the hamiltonian log density and its gradient , (b) the sampler and (c) the step size adaptor function.
```
Comment (Vaibhavdixit02) [inline docs/src/getting_started/bayesian_neural_ode_introduction.md:56]:
```suggestion
The AdvancedHMC interface requires us to specify: (a) the hamiltonian log density and its gradient , (b) the sampler and (c) the step size adaptor function.
```

**Comment by ChrisRackauckas** [comment]
This isn't posed correctly. "Bayesian neural ordinary differential equations" is not a "getting started" tutorial. It's not a tutorial you would expect to have your undergrad find as the third choice in a list when trying to learn Julia. The "getting started" tutorials are like 30 line snippets of code. This is a monster. It will scare people. It doesn't go there.

The applications page doesn't belong in docs. It is not documenting anything about the packages. It does deserve to go onto the website, which is why we have https://sciml.ai/showcase/ . And indeed that page should be updated with anything that it's missing, but we shouldn't have two different webpages about the showcase of how SciML has been used.

But okay, back to BNODE, what do we do here? Let's think about this page: https://docs.sciml.ai/Overview/stable/getting_started/getting_started/

When getting people started, there's two things to hit: how do you get started today, and what would your time investment in SciML get you in the long run? I think it what would be interesting would be tutorials focused on that latter part. So after "Introductory Tutorials", what about a section like "Some Advanced Tutorials to Show Off Cool Features"? Now that would allow you to dig in.

But even there, you shouldn't assume the audience is PhD researchers in a computational science domain. We'd need to up this to a level where even people who don't know the lingo still think it's cool. "Bayesian neural ODEs" assumes people know what Bayesian methods are and what neural ODEs are. So okay, we already lost 99% of potential people who are interested just by the title along. What can we do?

Let's hone in on the Lotka-Volterra one first. For that, "Auto-complete mechanistic models by embedding machine learning into differential equations". Then in the opening paragraph you describe that this is a UDE, here's the paper, why this would work, etc. etc., now let's dig into the code. Oh look how we spit out the LaTeX, all beautiful. Someone who has no prior knowledge of the field can get excited by that and get into start reading, and end with a "wow, that's possible"?

Next then a follow up: "Automated model discovery with quantified uncertainties and probability estimates". Now you link back to the first tutorial and say, read that one first. You then introduce the problem, the Bayesian neural ODE paper. You first show how to do it simple. We want code to be beautiful when possible. For this, just a pure Turing `@model` macro is all that's actually necessary to do this. But then we can say, but what about improving the sampling process? Then show digging into improving the samplers and all of that.

The purpose of the first tutorial is to show people the expanded idea of what SciML allows. The purpose of the second tutorial is to show how SciML connects with the greater universe (Turing) in a very tweakable way. With those purposes, the text design can then be simplified to hone in on those messages.

We need 5 like that, so we'll need to brainstorm. I think something along the lines of Brusselator MethodOfLines is a good one. Super practical and disgustingly easy. Something with structural identifiability and global sensitivity analysis together for examining parameter space since that's quite unique. And then the new GPU solvers outperforming JAX by 100x. That weaves a very strong narrative to make someone want to learn more for performance, curiosity, necessity for features ML folks ignore (I'm looking at you identifiability), and utility to PDEs.

**Comment by ChrisRackauckas** [comment]
Supersceded by https://github.com/SciML/SciMLDocs/pull/92. Your code is in that PR, though morphed into something more showcase-like.
